src/object_trajectory_predictor.hpp

```diff
@@ -40,11 +32,60 @@ class ObjectTrajectoryPredictor {
      *
      * This function calculates the new position of the object after, by user given, milliseconds
      *
-     * @param[in] The current position of the object, the speed of this object, the amount of milliseconds and the delay used
-     * between 2 measuring points while calculating speed
+     * @param[in] The current position of the object, the speed of this object, the amount of milliseconds
      * @return A vector object containing the new position of the object
      */
-    Vector3D calculatePositionAfterMs(Vector3D position, Vector3D speed, uint32_t ms, uint32_t measurementDelay);
+
+    Vector3D calculatePositionAfterMs(Vector3D position, Vector3D speed, uint32_t ms);
+
+  public:
+    ObjectTrajectoryPredictor();
+
+    /**
+     * @brief This method adds a new sample and updates speed.
+     *
+     *  This method adds a new sample to the internal stored sample array and calculates
+     *  the latest speed with the new sample array.
+     *  This function shifts the new sample into the position and measurementDelay arrays.
+     *  The position and delay are inserted at the end of the arrays. This is using the First in first out principle.
+     *  This method is assuming the sample size of the arrays is 10
+     *
+     *
+     * @param[in] a new sample as vector3D, the object id and the delay between last sample in milliseconds.
```

Comment:
Elke parameter moet zijn eigen @param[in] regel gevolgt met zijn naam.
Bijvoorbeeld:
@param[in]     newPosition     desc.
@param[in]     objectId            desc.
@param[in]     delayMs           desc.